src/components/Select.jsx

```diff
@@ -0,0 +1,213 @@
+import React from 'react';
+import { testBoard } from '../letters/Testboard';
+
+const SelectNumber = ({
+  numberSelected,
+  setNumberSelected,
+  setSelectNumber,
+  wordLengths,
+  setWordLengths,
+  squares,
+  setSquares,
+  wordNo,
+  setWordNo,
+  setErrorMessage,
+}) => {
+  const lengthList = [2, 3, 4, 5];
+  let useTestBoard = false;
+
+  const getRandomNumber = (start, end) => {
+    let random = Math.floor(Math.random() * end + start);
+    while (random > end) {
+      random = Math.floor(Math.random() * end + start);
+    }
+    return random;
+  };
+
+  const loadCell = (x, y, workSquares, doubleWord, workWordNo) => {
+    let newSquare = {};
+    let wordNums = [workWordNo];
+    if (doubleWord) {
+      wordNums = [workWordNo, workWordNo - 1];
+    }
+    newSquare = {
+      letter: '',
+      locationCol: x + ' / ' + (x + 1),
+      locationRow: y + ' / ' + (y + 1),
+      wordNums: wordNums,
+    };
+    workSquares.push(newSquare);
+    return workSquares;
+  };
+
+  const switchCell = workSquares => {
+    let savedLetter = workSquares[workSquares.length - 2].letter;
+    let savedWordNums = workSquares[workSquares.length - 2].wordNums;
+    workSquares[workSquares.length - 2].letter = workSquares[workSquares.length - 1].letter;
+    workSquares[workSquares.length - 2].wordNums = workSquares[workSquares.length - 1].wordNums;
+    workSquares[workSquares.length - 1].letter = savedLetter;
+    workSquares[workSquares.length - 1].wordNums = savedWordNums;
+    return workSquares;
+  };
+
+  const editInput = e => {
+    const value = e.target.value.replace(/[^0-9]/gi, '');
+    if (value > 20) return;
+    if (value === '0') {
+      setErrorMessage('Warning....number of letters selected is 0');
+    }
+    setNumberSelected(value);
+  };
+
+  function handleCheckboxChange(lth) {
+    let newWordLengths = [...wordLengths];
+    if (wordLengths.filter(item => item === lth).length > 0) {
+      newWordLengths = wordLengths.filter(item => item !== lth);
+    } else {
+      newWordLengths.push(lth);
+    }
+    setWordLengths(newWordLengths);
+  }
+
+  const loadBoard = () => {
```

Comment:
These kinda methods are the kind of methods that should go in your `useBoard` take all your board stuff and put it into that hook to hide it away from your UI code

Reply:
Luke.... should that include functions veriyboard and checkwords?

Reply:
I moved these 2 functions to utility.